lib/lita/handlers/totems.rb

```diff
@@ -240,15 +259,33 @@ def waiting_duration(time)
         ChronicDuration.output(Time.now.to_i - time.to_i, format: :short) || "0s"
       end
 
+      def take_totem(response, totem, user_id, timeout)
+        redis.set("totem/#{totem}/owning_user_id", user_id)
+        redis.sadd("user/#{user_id}/totems", totem)
+        redis.hset("totem/#{totem}/waiting_since", user_id, Time.now.to_i)
+        if @@DemoEnvironments.include? totem
+          # Create async job
+          after(timeout) do |timer|
```

Comment:
[question] what would happen with the timeouts if this service get restarted for whatever reason?

Reply:
the timeouts will be lost, they use threads to work

Reply:
Thanks. Then I was wondering if this would affect the stats in any way. Would we need to set anything on bootup?